Golang/Gorm/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+)
+
+type Employee struct {
+	gorm.Model
+	Name         string
+	Designation  string
+	DepartmentID uint
+	Department   Department
+	ManagerID    *uint
+	Manager      *Employee
+}
+
+type Department struct {
+	gorm.Model
+	Name      string
+	Employees []Employee
+}
+
+func main() {
+
+	db, err := gorm.Open("postgres", "user=postgres password=Reddy@1234 dbname=employees sslmode=disable")
+	if err != nil {
+		panic(err.Error())
+	}
+
+	defer db.Close()
+
+	seedDB(db)
+
+	employees := []Employee{}
+	db.Debug().Preload("Manager").Preload("Department").Find(&employees)
+
```

Comment:
Move logic into db/Employee.go

And might be the name could be ShowEmployeesDetails